lib/pacemaker/pcmk_cluster_queries.c

```diff
@@ -195,6 +195,7 @@ pacemakerd_event_cb(pcmk_ipc_api_t *pacemakerd_api,
             break;
 
         default:
+            data->rc = EBADMSG;
```

Comment:
We could make one or more new Pacemaker return codes for these. `EBADMSG` seemed close enough.

Reply:
With the pcmk_ipc_api_t model, a single callback is registered for all events, and it's up to the callback to filter out events it's not interested in. So, a different event type wouldn't be a failure, it would just be uninteresting. In other words, the callback isn't deactivated, and when the interesting event happens, the callback will see it.

The other places don't apply because we know what request we made and what reply we should expect. There's no such thing as an extraneous reply, replies match requests.

The other event type that can happen is a notification, which is not related to any request, but is a notification of something potentially interesting that happened on its own -- e.g. in the generic case of any daemon it could be that another client connected, the CIB was replaced, a node was fenced, etc.

Reply:
Okay, I think that makes sense. I was thinking we were done after we hit the callback once. But we keep listening until the timeout expires. Hopefully it's impossible to have a scenario like one client receiving a reply that's intended for another client... i.e., I hope it's guaranteed that a client receives at most one reply, and that that reply is intended for that client.

I guess there's also some mechanism for closing the client and deregistering its callback when it receives a reply. Otherwise with either mainloop or sync dispatch it would keep waiting (and receiving messages) until the timeout.

Reply:
Right, replies are targeted to the requester, while notifications would be broadcast.

For pcmk_cluster_queries.c, the relevant callbacks call event_done() to close the connection once a final answer is available.

Reply:
BTW just noticed `pcmk_cluster_queries.c:controld_event_reply()` already uses `EBADMSG`. I have no strong feelings; if you prefer `EPROTO`, I need to change it there too.

Reply:
It's not important -- they should never really happen. Either one conveys the idea.